modules/tts/tts-adapters/tts-adapter-sapinative/src/main/java/org/daisy/pipeline/tts/onecore/Onecore.java

```diff
@@ -10,7 +10,7 @@ public class Onecore {
 
 	public static native Voice[] getVoices() throws IOException;
 
-	public static native NativeSynthesisResult speak(String voiceVendor, String voiceName, String text) throws IOException;
+	public static native synchronized NativeSynthesisResult speak(String voiceVendor, String voiceName, String text) throws IOException;
```

Comment:
Could the `synchronized` possibly be conditional on whether we're running on Windows 11 or not?

Also, if the methods are synchronized, perhaps it would be better if the `reservedThreadNum` method of `SAPIEngine` would return 1?

Reply:
> Also, if the methods are synchronized, perhaps it would be better if the reservedThreadNum method of SAPIEngine would return 1?

Ha yes it could be an option too (i went to the fastest way to do it for tests, i did not looked into TTSengine superclass).

i'll try reducing the number of thread that way by checking for windows 11.
(I just need to check how to find out windows version)

Reply:
Note that the `reservedThreadNum` method can not a replacement of the `synchronized`. Even if the number of threads is set to 1, other TTS threads could still use the SAPI engine in case of a voice fallback. Also, multiple jobs could be running at the same time.

Reply:
So i did a small check : the system property are reporting the same "os.version" between windows 11 and windows 10 (which is 10.0)
But the os.name property returns "Windows 11" in my test, so i'll use this to check for windows version.

Reply:
Code has been updated to use a test on os.name property : 
For Onecore speak call, if os version ends with "11", the calls to the Onecore native method are synchonized on the SAPIEngine class object.

Just in case i tested with the legacy SAPI engine on windows 11 and the error does not occur with it in multithreading context on Windows 11 (the legacy engine is still present on the Windows 11 dev VM they provide) so it's only an issue within the latest version of their Onecore fork of their engine.

Reply:
OK, nice!

What about the `reservedThreadNum` idea? Could still be a good complementary solution I think.

Reply:
I can add it, but i don't know how it will change the current behavior
(or maybe it will reduce thread locking on the jvm side so that could reduce latency on job execution for this case ?)

Reply:
Note that this would also limit multithreading for SAPI legacy

Reply:
`reservedThreadNum` is now overriden to return 1 on Windows 11 and 0 otherwise

Reply:
> I can add it, but i don't know how it will change the current behavior

It might not change much in practice, especially when SAPI is the only TTS engine that is used. It's just that if you know in advance that only one sentence can be spoken at a time, it makes sense to do it all in one thread, in order to have as little overhead as possible. Probably not noticeable, but still.

> Note that this would also limit multithreading for SAPI legacy

Ah yes, I didn't think of that. Perhaps a little comment regarding this is appropriate.

Reply:
Added a comment